src/react/providers/room-provider.tsx

```diff
@@ -0,0 +1,94 @@
+import { RoomOptions, RoomOptionsDefaults } from '@ably/chat';
+import React, { ReactNode, useEffect, useState } from 'react';
+
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatClient } from '../hooks/use-chat-client.js';
+
+/**
+ * Props for the {@link RoomProvider} component.
+ */
+export interface RoomProviderProps {
+  /** The id of the room. */
+  id: string;
+
+  /**
+   * The options to use when creating the room.
+   *
+   * @defaultValue {@link RoomOptionsDefaults}
+   */
+  options?: RoomOptions;
+
+  /**
+   * Set to false to disable auto-releasing the room when component unmounts,
```

Comment:
I think we should set values like `false` to use the code markup, it looks a bit cleaner in typedocs IMO, what do you think? :)